cmake/Modules/SelectHashes.cmake

```diff
@@ -0,0 +1,65 @@
+# Select a hash backend
```

Comment:
I was planning to just stash whatever build support we need for hashes in that file as needed, hence the generic name. It could be easily morphed into a function/loop over hashes or something else.